<?xml version="1.0"?>
<launch>
    <arg name="sensor_model" default='VLP16' description="See readme for supported models"/>
    <arg name="return_mode" default="SingleStrongest" description="See readme for supported return modes"/>
    <arg name="frame_id"  default="velodyne"/>
    <arg name="scan_phase" default="0.0" description="Angle where scans begin (degrees, [0.,360.]" />

    <arg name="sensor_ip" default="192.168.1.201" description="Lidar Sensor IP"/>
    <arg name="host_ip" default="255.255.255.255" description="Broadcast IP from Sensor"/>
    <arg name="data_port" default="2368" description="LiDAR Data Port"/>
    <arg name="gnss_port" default="2369" description="LiDAR GNSS Port"/>
    <arg name="frequency_ms" default="100" description="LiDAR scan frequency"/>
    <arg name="packet_mtu_size" default="1500" description="Packet MTU size"/>

    <arg name="calibration_file" default="$(find-pkg-share nebula_lidar_driver)/calibration/velodyne/$(var sensor_model).yaml"/>
    <arg name="min_range" default="0.3" />
    <arg name="max_range" default="300.0" />
    <arg name="view_width" default="360.0" />

    <node pkg="nebula_lidar_driver" exec="velodyne_hw_ros_wrapper_node"
          name="velodyne_hw_driver" output="screen">
        <param name="sensor_model" value="$(var sensor_model)"/>
        <param name="return_mode" value="$(var return_mode)"/>
        <param name="frame_id" value="$(var frame_id)"/>
        <param name="scan_phase" value="$(var scan_phase)"/>
        <param name="sensor_ip" value="$(var sensor_ip)"/>
        <param name="frame_id" value="$(var frame_id)"/>
        <param name="host_ip" value="$(var host_ip)"/>
        <param name="data_port" value="$(var data_port)"/>
        <param name="gnss_port" value="$(var gnss_port)"/>
        <param name="frequency_ms" value="$(var frequency_ms)"/>
        <param name="packet_mtu_size" value="$(var packet_mtu_size)"/>
    </node>

    <node pkg="nebula_lidar_driver" exec="velodyne_driver_ros_wrapper_node"
          name="velodyne_cloud" output="screen">
        <param name="sensor_model" value="$(var sensor_model)"/>
        <param name="return_mode" value="$(var return_mode)"/>
        <param name="frame_id" value="$(var frame_id)"/>
        <param name="scan_phase" value="$(var scan_phase)"/>
        <param name="calibration_file" value="$(var calibration_file)"/>
        <param name="min_range" value="$(var min_range)"/>
        <param name="max_range" value="$(var max_range)"/>
        <param name="view_width" value="$(var view_width)"/>
    </node>

</launch>